Plugins/NativeEngine/Include/Babylon/ShaderCache.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <string_view>
+#include <memory>
+
+namespace Babylon
+{
+    class ShaderCache
+    {
+    public:
+        ShaderCache();
+        ~ShaderCache();
+
+        void SerializeOutTo(std::ofstream& stream);
+        uint32_t SerializeInFrom(std::ifstream& stream);
```

Comment:
Are these actually even used? It seems like the consuming code is actually just using the Impl as the public contract, even for serializing/deserializing.

Reply:
It should be at least tested in TestUnit.

Reply:
I've added a test.